pcdsdevices/epics/mps.py

```diff
@@ -134,3 +134,109 @@ def mps_factory(clsname, cls,  *args, mps_prefix, veto=False,  **kwargs):
     comp = FC(MPS, mps_prefix, veto=veto)
     cls  = type(clsname, (cls,), {'mps' : comp})
     return cls(*args, **kwargs)
+
+
+def mustBeOpenLogic(mps_A, mps_B):
+    """
+
+    This logic should analyze the two MPS classes of a device. This logic will only allow
+    beam through if the device is in the open-state.
+
+    Parameters
+    ----------
+    
+    mps_A.fault.value: Int
+    mps_B.fault.value: Int
+
+    Returns
+    -------
+    
+    bool
+        True if successful, False otherwise
+
+
+    """
+
+    if mps_A.fault.value == 1 and mps_B.fault.value == 1:
+        return False
+
+    if mps_A.fault.value == 0 and mps_B.fault.value == 0:
+        return False
+
+    if mps_A.fault.value == 1 and mps_B.fault.value == 0:
+        return False
+
+    if mps_A.fault.value == 0 and mps_B.fault.value == 1:
+        return True
+
+
+def mustKnowPositionLogic(mps_A, mps_B):
+    """
+
+    This logic should analyze the two MPS classes of a device. This logic will only allow
+    beam through if both the positions of the MPS classes is known
+
+    Parameters
+    ----------
+    
+    mps_A.fault.value: Int
+    mps_B.fault.value: Int
+
+    Returns
+    -------
+    
+    bool
+        True if successful, False otherwise
+    """
+    if mps_A.fault.value == 1 and mps_B.fault.value == 1:
+        return False
+
+    if mps_A.fault.value == 0 and mps_B.fault.value == 0:
+        return False
+
+    if mps_A.fault.value == 1 and mps_B.fault.value == 0:
+        return True
+
+    if mps_A.fault.value == 0 and mps_B.fault.value == 1:
+        return True
+
+
+class MPSLimits(Device):
+    """
+
+    The MPSLimits class is to determine what action is to be taken based on the MPS values
+    of a devicepertaining to a single device. If a device has two MPS values, there is 
+    certain logic that needs to be followed to determine whether or not the beam is allowed    through.
+
+    Parameters
+    ----------
+
+    Device: defined in MPS class above
+
+    Attributes
+    ----------
+
+    mps_A: the first MPS value of a Device
+
+    mps_B: the second MPS value of a Device
+   
+    name: str
+
+    logic: function
+        calls one of the previously defined functions based on the Device in question
+    """
+    mps_A = FC(MPS, '{self.MPSA}')
+    mps_B = FC(MPS, '{self.MPSB}')
+
+    def __init__(self, mps_A, mps_B, name=None, logic=None):
+
+        self.MPSA = mps_A
+        self.MPSB = mps_B
+        self.name = name
+        self.logic = logic
+        super().__init__('', name=name)
+
+    @property
+    def faulted(self):
+
```

Comment:
Add a docstring here that explains that we use the `logic` function to evaluate the faulted states of `mpsA` and `mpsB`